Neos.ContentRepository.Core/Classes/NodeType/NodeType.php

```diff
@@ -416,124 +158,75 @@ public function getNodeLabelGenerator(): NodeLabelGeneratorInterface
      * property configuration looks like.
      *
      * @return array<string,mixed>
-     * @api
+     * @deprecated with Neos 9.0 – {@see propertyDefinitions} should be used instead
      */
     public function getProperties(): array
     {
-        $this->initialize();
-
-        return ($this->fullConfiguration['properties'] ?? []);
-    }
-
-    /**
-     * Check if the property is configured in the schema.
-     */
-    public function hasReference(string $referenceName): bool
-    {
-        $this->initialize();
-
-        return isset($this->fullConfiguration['references'][$referenceName]);
-    }
-
-    /**
-     * Return the array with the defined references. The key is the reference name,
-     * the value the reference configuration. There are no guarantees on how the
-     * reference configuration looks like.
-     *
-     * @return array<string,mixed>
-     * @api
-     */
-    public function getReferences(): array
-    {
-        $this->initialize();
-
-        return ($this->fullConfiguration['references'] ?? []);
+        return $this->propertyDefinitions->map(function (PropertyDefinition $definition) {
+            return [
+                ...$definition->metadata,
+                'type' => $definition->type,
+                'defaultValue' => $definition->defaultValue,
+                'scope' => $definition->scope->value,
+            ];
+        });
     }
 
     /**
      * Check if the property is configured in the schema.
+     * @deprecated with Neos 9.0 – {@see propertyDefinitions::contain()} should be used instead
      */
     public function hasProperty(string $propertyName): bool
     {
-        $this->initialize();
-
-        return isset($this->fullConfiguration['properties'][$propertyName]);
+        return $this->propertyDefinitions->contain($propertyName);
     }
 
     /**
      * Returns the configured type of the specified property, and falls back to 'string'.
      *
-     * @throws \InvalidArgumentException if the property is not configured
+     * @deprecated with Neos 9.0 – {@see propertyDefinitions::get($propertyName)->type} should be used instead
      */
     public function getPropertyType(string $propertyName): string
     {
-        $this->initialize();
-
-        if (!$this->hasProperty($propertyName)) {
-            throw new \InvalidArgumentException(
-                sprintf('NodeType schema has no property "%s" configured for the NodeType "%s". Cannot read its type.', $propertyName, $this->name->value),
-                1708025421
-            );
+        $propertyDefinition = $this->propertyDefinitions->get(PropertyName::fromString($propertyName));
+        if ($propertyDefinition === null) {
+            throw new \InvalidArgumentException(sprintf('NodeType schema has no property "%s" configured for the NodeType "%s". Cannot read its type.', $propertyName, $this->name->value), 1708025421);
         }
```

Comment:
as this is now fully b/c we could consider reverting https://github.com/neos/neos-development-collection/pull/4480 so its fully 8.3 behaviour and we just return `string` always. That _would_ have the downside that for references it would then also return `string` but maybe we can also fake the old `reference|references` type as in our own code well use `propertyDefinitions::get($propertyName)->type` either way.

That _could_ make this method less breaking for 8.3 users

Reply:
You mean, we should return `'string'` for `$someNodeType->getPropertyType('unknownProperty')`??

Reply:
yes that was the 8.3 behaviour ... but i might as well take this on in a separate pr.
Im just trying to not have breaking changes everywhere but keep some parts functioning ^^

Reply:
I see, but IMO that behavior does not make sense **at all** :)